RELEASENOTES.md

```diff
@@ -13,9 +13,11 @@ Note that when using the preview version of AL-Go for GitHub, we recommend you U
 - Issue 997 'Deliver to AppSource' action fails for projects containing a space
 - Issue 987 Resource not accessible by integration when creating release from specific version
 - Issue 979 Publish to AppSource Documentation
+- Issue 1018 Artifact setting - possiblity to read version from app.json
 - Issue 1008 Allow PullRequestHandler to use ubuntu or self hosted runners for all jobs except for pregateCheck
 
 ### New Settings
+- `useEarliestArtifact`: set this to true in order for AL-Go for GitHub to build a project using the earliest compatible Business Central artifact, determined by the highest application dependency from app.json in all apps in the project and the `applicationDependency` in project settings.
```

Comment:
Maybe I misunderstand the original issue (or the name of the setting), but aren't they expecting to get the _latest_ artifact of a certain BC version rather than the earliest? 

If their artifact setting currently is `"artifact": "//23.1//latest"` I'm guessing they're expecting to get the latest 23.1 artifact

Reply:
That is the one I read 😄 Don't they want to use `"artifact": "//23.1//latest"` but without specifying 23.1

Reply:
"artifact": "//23.1//latest" will indeed give them the latest 23.1 - and it will continue to do so.
"artifact": "////latest" will give them the latest - artifact.
removing the artifact setting and setting useEarlistArtifact (a better name suggestion is welcome) will give them the first Business Central version they claim to be compatible with. If they have 23.1.52400.43123, 23.0.0.0, 20.1.0.0 and 22.5.23567.0 in 4 different App.Json's - they will get the first version greater or equal to 23.1.52400.43123. You could argue that we should take the latest 23.1 version, instead of looking at the full version number, but you could be in a situation where you would install on 23.1.52400.43123 right?

Reply:
By using the earliest we also get caching of the artifact - it doesn't change all the time.
Test Current can then test the current version.
In the end - we should build in a mechanism to clean up sandbox artifacts no longer used by tenants online - then we would build against exactly what we can end up installing this on.

Reply:
These comments gave me an idea though...
Instead of adding another property, we could use

"artifact": "//*//latest" for latest major with the same major.minor as the highest version from app.json
and
"artifact": "//*//first" for the first version -ge than the highest version from app.json

What do you think about that idea?

Reply:
> These comments gave me an idea though... Instead of adding another property, we could use
> 
> `"artifact": "//*//latest"` for latest major with the same major.minor as the highest version from app.json and `"artifact": "//*//first"` for the first version -ge than the highest version from app.json
> 
> What do you think about that idea?

I like that more 👍 

As I read the issue, this `"artifact": "//*//latest"` is really what the issue was asking for (maybe I misunderstood it though)

Reply:
That is exactly what he was asking for and `//*//first` was what -useEarliestArtifact would give you
I have changed the implementation and remove the useEarliestArtifact setting
Will run end 2 end to see if this has any impact

Reply:
The log will look like this:
![image](https://github.com/microsoft/AL-Go/assets/10775043/0a5d0eb1-47b5-4d46-8140-889ded4e4a10)